doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,48 @@ After programming and testing an application, you probably want to make some mod
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information about how the build system works in Zephyr, see :ref:`zephyr:build_overview` and :ref:`zephyr:application` pages in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr provides a CMake package that must be loaded by the application into its :file:`CMakeLists.txt` file.
+When loaded, the application is able to reference items provided by both Zephyr and the |NCS|.
+
+Loading Zephyr's `CMake`_ package creates the ``app`` CMake target.
+All files that your application uses can be added to this target.
+This can be done directly in the :file:`CMakeLists.txt` file or in the files that are sourced by it.
 
 To update the :file:`CMakeLists.txt` file, either edit it directly or use |SES| (SES) to maintain it.
 
 Editing CMakeLists.txt directly
 ===============================
 
-Add all files that your application uses to the ``target_sources`` function in :file:`CMakeLists.txt`.
-To include several files, it can be useful to specify them with a wildcard.
-For example, to include all :file:`.c` files from the :file:`src` folder, add the following lines to your :file:`CMakeLists.txt`::
+You must add all application source files to the ``app`` CMake target created by Zephyr.
+This can be done using the :c:func:`target_sources` function provided by CMake.
+You can use it directly in :file:`CMakeLists.txt`.
+
+Pay attention to the following configuration options:
+
+* If your application is complex, you can split it into subdirectories.
+  These subdirectories can provide their own :file:`CMakeLists.txt` files.
+* To include several files, you can specify them with a wildcard.
+  For example, to include all :file:`.c` files from the :file:`src` folder, add the following lines to your :file:`CMakeLists.txt`:
 
-   FILE(GLOB app_sources src/*.c)
-   target_sources(app PRIVATE ${app_sources})
+  .. code-block::
 
-Instead of specifying each file (explicitly or with a wildcard), you can include all files from a folder by adding that folder as include folder::
+     FILE(GLOB app_sources src/*.c)
+     target_sources(app PRIVATE ${app_sources})
 
-   target_include_directories(app PRIVATE src)
+* If your application provides header files, the build system will look for header files in include directories.
```

Comment:
```suggestion
* The build system searches for header files in include directories.
```

The build system will look for header files regardless of whether the application provides header files or not. But we need to introduce the concept of "include directories" for those developers not familiar with them.